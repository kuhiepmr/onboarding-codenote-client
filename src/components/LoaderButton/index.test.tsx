import { shallow } from 'enzyme';
import React from 'react';
import LoaderButton from './index';

test('renders without crashing', () => {
  const props = {
    isLoading: true,
    loadingText: 'loading',
    text: 'button'
  };
  const wrapper = shallow(<LoaderButton {...props} />);

  expect(wrapper).toMatchSnapshot();
});

describe('LoaderButton', () => {
  it('should render loadingText', () => {
    const props = {
      isLoading: true,
      loadingText: 'loading',
      text: 'button'
    }
    const wrapper = shallow(<LoaderButton {...props} />);

    expect(wrapper.dive().find('button').text()).toBe('<Glyphicon />loading')
  });

  it('should render text', () => {
    const props = {
      isLoading: false,
      loadingText: 'loading',
      text: 'button'
    }
    const wrapper = shallow(<LoaderButton {...props} />);

    expect(wrapper.dive().find('button').text()).toBe('button')
  });

});