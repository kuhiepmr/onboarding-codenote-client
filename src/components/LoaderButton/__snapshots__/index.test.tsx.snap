// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="LoaderButton "
  disabled={true}
>
  <Glyphicon
    bsClass="glyphicon"
    className="spinning"
    glyph="refresh"
  />
  loading
</Button>
`;
